# HG changeset patch
# Parent  23cfafefa564ef94d4b60d3e157dd8ff58b2b5c5
Properly do type punning on float endian conversion functions

This only showed up with optimizations enabled.

diff --git a/lifx/wire_proto.h b/lifx/wire_proto.h
--- a/lifx/wire_proto.h
+++ b/lifx/wire_proto.h
@@ -28,15 +28,17 @@
 static inline floatle_t
 lgtd_lifx_wire_htolefloat(float f)
 {
-    *(uint32_t *)&f = htole32(*(uint32_t *)&f);
-    return f;
+    union { float f; uint32_t i; } u  = { .f = f };
+    htole32(u.i);
+    return u.f;
 }
 
 static inline floatle_t
 lgtd_lifx_wire_lefloattoh(float f)
 {
-    *(uint32_t *)&f = le32toh(*(uint32_t *)&f);
-    return f;
+    union { float f; uint32_t i; } u  = { .f = f };
+    le32toh(u.i);
+    return u.f;
 }
 
 enum { LGTD_LIFX_PROTOCOL_PORT = 56700 };
